### ISSUE TITLE
Question about sending BLE broadcast to trigger visibility

### ISSUE THREAD

**Issue by ciriousjoker** [grishka/NearDrop#109]
I have read the FAQ and it states that it's not possible because Apple doesn't allow it. I'm wondering why it's not possible to bypass it and I have a couple of questions. Keep in mind I have not much of a clue about Bluetooth, I'm just trying to understand this better. If you don't have time/don't want to answer, feel free to close this issue.

- I assume it's denied in CoreBluetooth? I couldn't find anything about this, but this post suggests [sending broadcasts](https://diamantidis.github.io/2020/02/16/broadcast-and-scan-nearby-devices-using-corebluetooth) is possible.
- What about IOBluetooth? According to [this guy](https://stackoverflow.com/a/68794112), it apparently doesn't support BLE, but he suggests it can do very low level discovery. Perhaps it could be enough to send broadcasts?

This could be bypassed by sending push notifications to a device that has a companion app installed, but I assume this isn't planned? If so, I can help with that part, but since you say you're and Android developer, I'm sure you know how to do this stuff and probably just decided it's not worth the hassle.

**Comment by grishka** [comment]
I tried using different APIs, public and private. Both CoreBluetooth and IOBluetooth. I found [an example that purportedly sends raw HCI commands to the Bluetooth controller using private APIs](https://colemancda.github.io/2018/03/25/Hacking-IOBluetooth), but it didn't work on macOS Ventura. I even disassembled IOBluetooth but most of the private functions this thing was calling were no-op (just the `ret` instruction) and the one that wasn't was returning an error. It might have worked on previous system versions but yeah it's Apple, if you use a private API and your program breaks, you get to keep both parts.

> This could be bypassed by sending push notifications to a device that has a companion app installed, but I assume this isn't planned?

This would require a second device and an app to be installed on it. The whole point of NearDrop is that it's a *self-sufficient* implementation of Nearby Share for macOS.
